src/sentry/utils/suspect_resolutions/commit_correlation.py

```diff
@@ -2,25 +2,21 @@
 
 from typing import Set
 
-from sentry.models import CommitFileChange, GroupRelease, ReleaseCommit
+from sentry.models import CommitFileChange, Group, GroupRelease, Project, ReleaseCommit
 
 
-def is_issue_commit_correlated(
-    resolved_issue_id: int, candidate_issue_id: int, project_id: int
-) -> bool:
-
-    resolved_issue_files = get_files_changed(resolved_issue_id, project_id)
-    candidate_suspect_resolution_files = get_files_changed(candidate_issue_id, project_id)
+def is_issue_commit_correlated(resolved_issue: int, candidate_issue: int, project: int) -> bool:
```

Comment:
A row in `GroupRelease` is created when there’s an event within a specific group and release, so `GroupRelease` gives us all the releases that a group has ever been seen in. I don’t think we need to split up issues that are resolved in a release/commit/pr and issues that are resolved manually - we can just query for the release(s) that the issue is tied to in all cases. The limitation here is that we can only check for files changed in issues that are tied to at least one release.

Reply:
I think this makes sense here. Commit correlations is really only there to cull down the number of correlations we'll be doing. Even if we have some aggressive commits correlated between issue-releases, that should still be ok. The metric correlation should be the deciding factor whether we surface up a suspect resolution.